opentreemap/treemap/js/src/searchBar.js

```diff
@@ -224,18 +238,63 @@ function updateDisabledFieldGroups(search) {
             _(fieldGroupsToEnable).pull('EmptyPlot').push('Plot');
         }
         $(dom.fieldGroup).addClass('disabled');
-        $(dom.fieldsDisabledMessage).show();
+        $(dom.fieldGroupDisabledMessage).show();
         _.each(fieldGroupsToEnable, function (featureName) {
             var $group = $('#search-fields-' + featureName);
             $group.removeClass('disabled');
-            $group.find(dom.fieldsDisabledMessage).hide();
+            $group.find(dom.fieldGroupDisabledMessage).hide();
         });
     } else {
         $(dom.fieldGroup).removeClass('disabled');
-        $(dom.fieldsDisabledMessage).hide();
+        $(dom.fieldGroupDisabledMessage).hide();
     }
 }
 
+function updateDisabledFields(search) {
+    var minMax = ['MIN', 'MAX'];
+
+    // First enable all search fields
+    $(dom.searchFields).prop('disabled', false);
+    $(dom.fieldDisabledMessage).hide();
+    $(dom.searchFieldContainer).removeClass('disabled');
+    updateDisabledSpeciesFields(false);
+
+    // Then disable all fields which are not filled in but which have the same
+    // key as another filled in field
+    _.each(search.filter, function(predicate, field) {
+        var searchTypes = _.keys(predicate),
+            minOrMax = _.contains(searchTypes, 'MIN') || _.contains(searchTypes, 'MAX');
+        $(dom.searchFields).filter('[name="' + field + '"]').each(function(i, elem) {
+            var $elem = $(elem),
+                searchType = $elem.attr('data-search-type');
+
+            // Min/Max fields don't affect other min/Max fields
```

Comment:
I didn't get this on first reading. Maybe something like "e.g. when user fills in the min field, don't disable the corresponding max field."